lib/pengine/clone.c

```diff
@@ -882,12 +882,17 @@ pe__clone_html(pcmk__output_t *out, va_list args)
             list = g_list_sort(list, sort_node_uname);
             for (nIter = list; nIter != NULL; nIter = nIter->next) {
                 pe_node_t *node = (pe_node_t *)nIter->data;
+                GList *node_list = NULL;
 
-                if (pe_find_node(rsc->running_on, node->details->uname) == NULL &&
+                node_list = g_list_prepend(node_list, (gpointer) node->details->uname);
+
+                if (!pe__rsc_running_on_any_node_in_list(rsc, node_list) &&
                     pcmk__str_in_list(only_node, node->details->uname)) {
                     pcmk__add_word(&stopped_list, &stopped_list_len,
                                    node->details->uname);
                 }
+
+                g_list_free(node_list);
```

Comment:
With this new approach I'm thinking you were onto something with a wrapper for the single-node case, but for g_list_prepend()+pe__rsc_...()+g_list_free(). That would be very close to pe_find_node() at that point

Reply:
Currently, there's only two places this would be useful and they're both in lib/pengine/clone.c.  I'll add a static wrapper in there and if it becomes more useful later on, we can always move it out.